### ISSUE TITLE
Enable ipv6 client side support by default

### ISSUE THREAD

**Issue by stevie2sheds** [jamulussoftware/jamulus#3300]
**Current behaviour and why should it be changed?**

IPv6 client support currently has to be enabled at the command line using the -6 option. This requires the user either to edit the shortcuts/links or start the client from the command line. This varies in complexity depending on the OS, but it is a major barrier to IPv6 adoption when trying to support large groups having to make the required changes to client startup. Having IPv6 support enabled by default would remove this barrier.

**Describe possible approaches**
Another approach could be to enable IPv6 support using a checkbox within the UI. This would allow the user to elect to enable IPv6, and perhaps there is a very marginal resource saving in not starting a feature that may not be used. 

**Describe the solution you'd like**
Enable ipv6 client side support by default with no additional checkbox in the UI. That solution means no action needs to be taken at the client side at all for IPv6.

**Has this feature been discussed and generally agreed?**

The topic has been discussed here: [https://github.com/orgs/jamulussoftware/discussions/3291](https://github.com/orgs/jamulussoftware/discussions/3291)

**Comment by softins** [comment]
If we did make it a checkbox in Settings, it should also be stored in the ini file, to remember the setting.

In that case, rather than the complication of changing the open socket on the fly when the checkbox is toggled, it may well be sufficient to request a restart, like it does when the language setting is changed.

Would also need to make sure the ini file is loaded before the socket is created.

**Comment by stevie2sheds** [comment]
That sounds as if the checkbox option is of greater complexity than enabling by default and therefore the always on option is preferable.

**Comment by pljones** [comment]
I've tagged this as Release 4.0.0 as it'll probably get picked up alongside #3242.

**Comment by ann0see** [comment]
Would we be confident enough to not even provide an IPv6 off switch and rely on `--serverbindip` to disable IPv6 or 4 if needed?

**Comment by softins** [comment]
> Would we be confident enough to not even provide an IPv6 off switch and rely on `--serverbindip` to disable IPv6 or 4 if needed?

I would agree with making IPv6 enabled the default for clients, but I think for servers it should still be down to the server operator to enable it with `-6`. Thus making `-6` into a server-only option rather than both server and client.

**Comment by ann0see** [comment]
I do not agree with default disabled. It should be default enabled for server and client. In case of problems and only then one may argue to disable IPv6. Otherwise it'll never take off.

**Comment by softins** [comment]
We should test what happens currently when specifying `-6` on a host that doesn't have an IPv6 address.

**Comment by softins** [comment]
> We should test what happens currently when specifying `-6` on a host that doesn't have an IPv6 address.

```
$ ./Jamulus -s -n -p 22120 -6 -d -F -u 5 -o 'PiTest;Winchester;224' -w 'Testing only' -e localhost
- server mode chosen
- no GUI mode chosen
- selected port number: 22120
- IPv6 enabled
- disconnect all clients on quit
- using 64 samples frame size mode
- maximum number of channels: 5
- server info: PiTest;Winchester;224
- welcome message: Testing only
- register with directory at address: localhost
Jamulus: Network Error: IPv6 requested but not available on this system.
$
```

So we need to make it automatically fall back to IPv4 only before we can make IPv6 enabled by default.

**Comment by pljones** [comment]
> So we need to make it automatically fall back to IPv4 only before we can make IPv6 enabled by default.

No, the user in that example explicitly stated they wanted IPv6 only.  It would be wrong to fall back to IPv4, in my view.  It's like skipping "Don't show the UI" and then deciding because they've not got a desktop environment to run in headless mode anyway.  They won't know what's going on.

If we want a separate "and fall back to IPv4" flag, I'd be happier with that.  A single argument shouldn't conflate things.

**Comment by ann0see** [comment]
Hmm. So:
-6 -> require IPv6 on no IPv6 fail. 
No -6 default IPv6 and 4 no IPv6 -> fallback with warning

**Comment by pljones** [comment]
Yeah, I'd be happy with that.  We could add `-4` for "don't even try IPv6 as it causes me problems" cases.

**Comment by softins** [comment]
> > So we need to make it automatically fall back to IPv4 only before we can make IPv6 enabled by default.
> 
> No, the user in that example explicitly stated they wanted IPv6 only.

That's not what `-6` means currently.

In all versions since sometime in 2021, the behaviour has been as follows:

| Option | Behaviour |
|:---:|:--- |
| omit `-6` | Use IPv4 only (the default) |
| add `-6` | Use a dual-stack socket, allowing both IPv4 and IPv6. Currently fails if the host doesn't support IPv6 |

There is currently no way to say "only use IPv6".

> It would be wrong to fall back to IPv4, in my view. It's like skipping "Don't show the UI" and then deciding because they've not got a desktop environment to run in headless mode anyway. They won't know what's going on.
> 
> If we want a separate "and fall back to IPv4" flag, I'd be happier with that. A single argument shouldn't conflate things.

So we need to agree specifically, for both client and server, whether we want dual-stack to be the default, and then whether we want options to specify IPv4-only (e.g. `-4`) and/or IPv6-only (e.g. `-6`, but this would be a change of meaning), and presumably would then reject with an error if the user tried specifying both options.

And if neither option was specified, and the system does not support IPv6, it should default to IPv4 anyway, as at present (without needing a warning).

**Comment by ann0see** [comment]
> There is currently no way to say "only use IPv6"

What happens if you bind to an IPv6 address using `--serverbindip`?

**Comment by ann0see** [comment]
Dual stack default is imo the most sensible thing to do. 

My main objective is that IPv6 is enabled by default if the server has an IPv6 address. The admin should never need to IPv6.

**Comment by softins** [comment]
> > There is currently no way to say "only use IPv6"
> 
> What happens if you bind to an IPv6 address using `--serverbindip`?

Currently, if `-6` is specified, the argument to `--serverbindip` has no effect:
https://github.com/jamulussoftware/jamulus/blob/7187813b92ef0619c04e07008cc1f2d300deadb2/src/socket.cpp#L105-L185

**Comment by stevie2sheds** [comment]
Many thanks team for all your work on this and the project.
